common/src/main/java/org/wso2/testgrid/common/DashboardSetup.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.util.StringUtil;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * This class create a database in influxDB and a Data Source in
+ */
+public class DashboardSetup {
+
+    private static final Logger logger = LoggerFactory.getLogger(DashboardSetup.class);
+    private String restUrl = "http://ec2-54-167-98-19.compute-1.amazonaws.com";
+    private String username = "wso2";
+    private String password = "wso2123";
+    private String apikey = "Bearer eyJrIjoiZlZlVDNWMXZFbzdZdGp6R3NMaWlvdmRaRE5sYjFuWXkiLCJuIjoid3NvMiIsImlkIjoxfQ==";
+    private String testplanID;
+
+    public DashboardSetup(String testplanID) {
+        this.testplanID = testplanID;
+    }
+
+    /**
+     * this method will create a DB in influxDB and add a new Data source to grafana
+     */
+    public void initDashboard() {
+        // create influxDB database according to tp_id
+        try {
+            InfluxDB influxDB = InfluxDBFactory.connect(restUrl + ":8086", username, password);
+            String dbName = testplanID;
+            influxDB.createDatabase(dbName);
+            influxDB.close();
+        } catch (AssertionError e) {
+            logger.error(StringUtil.concatStrings("Cannot create a new Database: ", e));
+        }
+        // add a new data source to grafana
+        HttpPost httpPost = createConnectivity(restUrl, apikey);
+        executeReq(getDataSource(testplanID), httpPost);
+
+    }
+    /**
+     * This method creates a json string that describes the grafana datasource
+     *
+     * @param name name of the data source that need to be created which is test plan ID
+     * @return string of jason string of the data source
+     */
+
+    private String getDataSource(String name) {
+
+        JSONObject user = new JSONObject();
+        user.put("name", name);
+        user.put("type", "influxdb");
+        user.put("url", "http://localhost:8086");
+        user.put("access", "proxy");
+        user.put("basicAuth", false);
+        user.put("password", "wso2123");
+        user.put("user", "telegraf");
+        user.put("database", name);
+
+        return user.toString();
+
+    }
+
+    /**
+     * this method will create the header of the POST request
+     * @param restUrl url of the grafana server
+     * @param apikey APIkey of the grafana server
+     * @return return a http post request
+     */
+
+    private HttpPost createConnectivity(String restUrl, String apikey) {
+        HttpPost post = new HttpPost(restUrl + ":3000/api/datasources/");
+        post.setHeader("AUTHORIZATION", apikey);
+        post.setHeader("Content-Type", "application/json");
+        post.setHeader("Accept", "application/json");
```

Comment:
Use existing constants available for content-type. If not define a new constant.